### PR TITLE
Turn federated auth on for Identity tests.

### PR DESCRIPTION
**Pull request by ahsonkhan**
Follow-up from https://github.com/Azure/azure-sdk-for-cpp/pull/5740#discussion_r1659213420

The liveonly test is being disabled with tracking issue https://github.com/Azure/azure-sdk-for-cpp/issues/5786

### CODE REVIEW

**Review by antkmsft — APPROVED** [review]

File: sdk/identity/ci.yml [hunk 1]
```diff
@@ -30,6 +30,7 @@ extends:
       LiveTestCtestRegex: azure-identity.
       LineCoverageTarget: 95
       BranchCoverageTarget: 56
+      UseFederatedAuth: true
```
Comment (RickWinter) [inline sdk/identity/ci.yml:33]: @gearama Is this necessary? It was my understanding that federated auth is on by default in the pipelines and you only need to set this to false when opting out

Comment (ahsonkhan) [reply sdk/identity/ci.yml:33]: That doesn't seem to be true, see similar change in ci.yml here:
https://github.com/Azure/azure-sdk-for-cpp/pull/5754/files#diff-126f4d961feeee67d4686c25c0682043833c8b596697eb2117f9da8735cbf38bR33

And @gearama's response to your question:
https://github.com/Azure/azure-sdk-for-cpp/pull/5754#discussion_r1668978160

> RickWinter: Is this necessary if default is true?
> gearama: yes

**Comment by ahsonkhan** [comment]
The liveonly test is being disabled with tracking issue https://github.com/Azure/azure-sdk-for-cpp/issues/5786

**Pull request merged**